2024/XenSummit/Challenges_and_Status_of_Enabling_TrenchBoot_in_Xen_Hypervisor.md

```diff
@@ -0,0 +1,234 @@
+
+- Aims for unified approach supporting both AMD and Intel
+  processors
+- The goal is to implement a native support for D-RTM to let Xen have full
+  control without any exokernels
+]
+
+???
+
+Some may know about tboot - Intel TXT reference implementation
+
+With native support for DRTM, there are some additional steps to be performed
+before the CPUs may be initialized (wait for SIPI state). There is also a
+matter of handling power state transitions and event log.
+
+---
+
+# Usecase: Qubes OS Anti Evil Maid
+
+.center[.image-30[![](/img/qubes-logo-home.svg)]]
+
+- Qubes OS Anti Evil Maid (AEM) is a set of software packages and utilities to
+  aid against [Evil Maid attacks](https://theinvisiblethings.blogspot.com/2009/10/evil-maid-goes-after-truecrypt.html)
+- Leverages DRTM and TPM to seal secrets, which are used by the owner to
+  confirm whether the device has been tampered with or not
+- Initially only Intel TXT and TPM1.2 with tboot was supported using legacy
+  BIOS boot mode, now TrenchBoot replaces tboot, extending the support to
+  TPM2.0 and AMD SKINIT
+
+---
+
+# Demo: Qubes OS Anti Evil Maid
+
+.center[ <a href="https://asciinema.org/a/661348?cols=96&rows=24"
+target="_blank"> <img src="https://asciinema.org/a/661348.svg" width="400px"/>\
+</a> ]
+
+.footnote[
+Watch on [asciinema](https://asciinema.org/a/661348?cols=96&rows=24)
+]
+
+---
+
+# Current status of the work
+
+- Added TPM 2.0 support for dom0 kernel and initrd measurements
+- Added eventlog extraction and parsing
+- Rebased changes onto the most recent TrenchBoot support patches to Linux
+  kernel introducing a unified TrenchBoot boot protocol
+- Final shape of patches is not yet known, the work on finalizing the boot
+  protocol including UEFI is pending on the [Linux kernel mailing list,
+  currently patchset
+  v8](https://lore.kernel.org/lkml/8d543a15-af62-4403-b2e0-3b395edfe9e4@amd.com/T/)
+- Simplified installation of the required packages by providing a RPM
+  repository for QubesOS
+- Implemented support for booting with Intel TXT and TPM 2.0 in legacy BIOS
+  boot mode
+- Implemented support for booting with AMD SKINIT and TPM 1.2/2.0 in legacy
+  BIOS boot mode
+
+---
+
+# Current ongoing efforts
+
+- Provide package repositories for other major distros (Debian, Ubuntu Fedora)
+- Prepare tools for assessing the readiness of the system and BIOS to perform
+  Dynamic Launch
+- Adding DRTM Ready indicator to the system security reports: QubesOS Security
+  Report, [fwupd HSI](https://fwupd.github.io/libfwupdplugin/hsi.html)
+
+.center[.image-40[![](/img/qubes-sec-report.png)]]
+
+---
+
+# Try TenchBoot
+
+It is relatively easy to get a hardware which supports DRTM:
+
+- Intel-based tested and known to work:
+  * Protectli VP4670
+  * HP EliteDesk 800 G2 Desktop Mini
+- Any Intel vPro laptop/desktop should generally work, but DRTM is sensitive
+  to BIOS bugs related to Intel TXT initialization (some platforms simply do
+  not set it up properly)
```

Comment:
How do you verify that the hardware would work?

Reply:
keyword SHOULD, you won't know unless you actually try it out

Reply:
We cannot accept that answer on engineering and business grounds. We should write down all aspects of SBoM and HBoM that can affect whether this feature will work or not. Then, prepare a test for each aspect that can verify if the given property is present.  This is very important.

Obviously, we cannot guarantee anything since we cannot validate all possible configurations, but at least we can provide such validation results for the platforms we tried.

This presentation is part of building awareness that we are experts in TrenchBoot enabling.

Reply:
Rephrased it: https://github.com/3mdeb/conferences/pull/31/commits/fe1f444b41f5daf3a9278f4d71082fffc51e7dca#diff-28488ef9a30ac840715c1e53410d8d69b64e939b4fd9fc4ea63380479fc4dc43R263-R273